src/component/archive/ArchiveTable/ArchiveTable.tsx

```diff
@@ -0,0 +1,338 @@
+import { PageContent } from 'component/common/PageContent/PageContent';
+import { PageHeader } from 'component/common/PageHeader/PageHeader';
+import {
+    SortableTableHeader,
+    Table,
+    TableBody,
+    TableCell,
+    TablePlaceholder,
+    TableRow,
+    TableSearch,
+} from 'component/common/Table';
+import { useFlexLayout, useSortBy, useTable } from 'react-table';
+import { SearchHighlightProvider } from 'component/common/Table/SearchHighlightContext/SearchHighlightContext';
+import { useMediaQuery } from '@mui/material';
+import { sortTypes } from 'utils/sortTypes';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { HighlightCell } from 'component/common/Table/cells/HighlightCell/HighlightCell';
+import { DateCell } from 'component/common/Table/cells/DateCell/DateCell';
+import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+import { FeatureTypeCell } from '../../common/Table/cells/FeatureTypeCell/FeatureTypeCell';
+import { FeatureSeenCell } from '../../common/Table/cells/FeatureSeenCell/FeatureSeenCell';
+import { LinkCell } from '../../common/Table/cells/LinkCell/LinkCell';
+import { FeatureStaleCell } from '../../feature/FeatureToggleList/FeatureStaleCell/FeatureStaleCell';
+import { ReviveArchivedFeatureCell } from 'component/archive/ArchiveTable/ReviveArchivedFeatureCell/ReviveArchivedFeatureCell';
+import { useStyles } from '../../feature/FeatureToggleList/styles';
+import { featuresPlaceholder } from '../../feature/FeatureToggleList/FeatureToggleListTable';
+import theme from 'themes/theme';
+import { FeatureSchema } from '../../../openapi';
+import { useFeatureArchiveApi } from 'hooks/api/actions/useFeatureArchiveApi/useReviveFeatureApi';
+import useToast from 'hooks/useToast';
+import { formatUnknownError } from 'utils/formatUnknownError';
+import { useSearch } from 'hooks/useSearch';
+import { FeatureArchivedCell } from '../../common/Table/cells/FeatureArchivedCell/FeatureArchivedCell';
+import { useVirtualizedRange } from 'hooks/useVirtualizedRange';
+
+export interface IFeaturesArchiveTableProps {
+    archivedFeatures: FeatureSchema[];
+    title: string;
+    refetch: any;
```

Comment:
This should have static type. Same with `storedParams`, `setStoredParams`, `searchParams` and `setSearchParams`.